frontend/app/src/components/history/LocationIcon.vue

```diff
@@ -14,10 +15,16 @@ const props = defineProps({
   noPadding: { required: false, type: Boolean, default: false }
 });
 
-const { size } = toRefs(props);
+const { item, size } = toRefs(props);
 const iconStyle = computed(() => ({
   fontSize: get(size)
 }));
+
+const { isPending } = useAssetCacheStore();
+const loading: ComputedRef<boolean> = computed(() => {
+  const itemVal = get(item);
+  return !itemVal.exchange && get(isPending(itemVal.identifier));
+});
```

Comment:
Can I have a brief explanation on why and where do we need these changes?

At least from what I remember on how this component works this doesn't make much sense to me.